cookbooks/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+# Copyright (C) 2023 by the authors of the ASPECT code.
+#
+# This file is part of ASPECT.
+#
+# ASPECT is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+#
+# ASPECT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ASPECT; see the file doc/COPYING.  If not see
+# <http://www.gnu.org/licenses/>.
+
+CMAKE_MINIMUM_REQUIRED(VERSION 3.1.0)
+
+MESSAGE(STATUS "===== Configuring ASPECT cookbooks =============")
+
+####################################################
+# Generate targets for the cookbooks
+####################################################
+
+# Step 1: Find all of the cookbooks that have separate plugins.
+FILE(GLOB_RECURSE _cmakelists_cookbooks
+     ${CMAKE_SOURCE_DIR}/cookbooks/**/CMakeLists.txt)
+
+# Step 2: Add all subdirectories of plugins.
+FOREACH(_file ${_cmakelists_cookbooks})
```

Comment:
I would use generic names (the 2 CMakeLists.txt files can basically be identical)